File: user-service/user-core/src/main/java/com/yellowsunn/userservice/domain/BaseTimeEntity.java
package com.yellowsunn.userservice.domain;

import jakarta.persistence.MappedSuperclass;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import lombok.Getter;

import java.time.ZonedDateTime;

@Getter
@MappedSuperclass
public abstract class BaseTimeEntity {
    private ZonedDateTime createdAt;
    private ZonedDateTime modifiedAt;

    @PrePersist
    void prePersist() {
        var now = ZonedDateTime.now();
        this.createdAt = now;
        this.modifiedAt = now;
    }

    @PreUpdate
    void preUpdate() {
        this.modifiedAt = ZonedDateTime.now();
    }
}

File: board-service/libs/board-core/src/main/kotlin/com/yellowsunn/boardservice/domain/command/BaseTimeEntity.kt
package com.yellowsunn.boardservice.domain.command

import jakarta.persistence.MappedSuperclass
import jakarta.persistence.PrePersist
import jakarta.persistence.PreUpdate
import java.time.ZonedDateTime

@MappedSuperclass
abstract class BaseTimeEntity {
    lateinit var createdAt: ZonedDateTime
        private set

    lateinit var modifiedAt: ZonedDateTime
        private set

    @PrePersist
    fun prePersist() {
        val now = ZonedDateTime.now()
        this.createdAt = now
        this.modifiedAt = now
    }

    @PreUpdate
    fun preUpdate() {
        this.modifiedAt = ZonedDateTime.now()
    }
}
